pydicom/values.py

```diff
@@ -388,7 +389,8 @@ def convert_numbers(
     if length % bytes_per_value != 0:
         raise BytesLengthException(
             "Expected total bytes to be an even multiple of bytes per value. "
-            f"Instead received {byte_string!r} with length {length} and "
+            f"Instead received {shorten(repr(byte_string), width=10_000)} "
+            f"with length {length} and "
             f"struct format '{struct_format}' which corresponds to bytes per "
```

Comment:
Thanks for this, it's a good idea to clip this output somehow.
  
But when I try `shorten` on some binary values I always get "[...]", so it's not very helpful.
E.g.
```python
b = b"\5" * 100
shorten(repr(b), 50)
'[...]'
```

Also, I think 10_000 is still too large to display and be helpful.

I would suggest if `length` is less than say, a few hundred, print the whole repr, otherwise the message can skip the `repr(byte_string)` and just print the length.

Reply:
Updated, how's that?